test/components.js

```diff
@@ -316,6 +317,131 @@ test('disable raw mode when all input components are unmounted', t => {
 	t.true(stdin.pause.calledOnce);
 });
 
+test('setRawMode should throw if not supported', t => {
+	const stdout = {
+		write: spy(),
+		columns: 100
+	};
+
+	const stdin = new EventEmitter();
+	stdin.setEncoding = () => {};
+	stdin.setRawMode = spy();
+	stdin.isTTY = false; // Without this, setRawMode will throw
+	stdin.resume = spy();
+	stdin.pause = spy();
+
+	const didCatchInMount = spy();
+	const didCatchInUnmount = spy();
+
+	const options = {
+		stdout,
+		stdin,
+		debug: true
+	};
+
+	class Input extends React.Component {
+		render() {
+			return <Box>Test</Box>;
+		}
+
+		componentDidMount() {
+			try {
+				this.props.setRawMode(true);
+			} catch (error) {
+				didCatchInMount(error);
+			}
+		}
+
+		componentWillUnmount() {
+			try {
+				this.props.setRawMode(false);
+			} catch (error) {
+				didCatchInUnmount(error);
+			}
+		}
+	}
+
+	const Test = () => (
+		<StdinContext.Consumer>
+			{({setRawMode}) => (
+				<Input setRawMode={setRawMode}/>
+			)}
+		</StdinContext.Consumer>
+	);
+
+	const {unmount} = render(<Test/>, options);
+	unmount();
+
+	t.is(didCatchInMount.callCount, 1);
+	t.is(didCatchInUnmount.callCount, 1);
+	t.false(stdin.setRawMode.called);
+	t.false(stdin.resume.called);
+	t.false(stdin.pause.called);
+});
+
+test('isRawModeSupported should be used if setRawMode is not supported', t => {
```

Comment:
I don't think I understand the test title here, what is test about?

Reply:
It makes sure that components using the isRawModeSupported properly doesn't result in errors thrown. Thanks for the note 👍